vk_video_decoder/libs/NvVideoParser/src/VulkanAV1Decoder.cpp

```diff
@@ -2245,29 +2245,34 @@ bool VulkanAV1Decoder::ParseObuTileGroup(const AV1ObuHeader& hdr)
 	// Tile payload
     int consumedBytes = (consumed_bits() + 7) / 8;
 	//                                                    offset of obu         number of bytes read getting the tile data
-	m_PicData.tileOffsets[m_PicData.khr_info.tileCount] = m_nalu.start_offset + consumedBytes;
 
-	// Compute the tile group size
-	uint32_t totalTileSize = 0;
     for (int TileNum = tg_start; TileNum <= tg_end; TileNum++)
     {
         int lastTile = TileNum == tg_end;
         size_t tileSize = 0;
+
         if (lastTile)
         {
             tileSize = hdr.payload_size - consumedBytes;
+            m_PicData.tileOffsets[m_PicData.khr_info.tileCount] = m_nalu.start_offset + consumedBytes;
```

Comment:
There seem to be a few implicit narrowcasting issues with the new code. The tileoffset are 32b while start_offset is 64bit.
Probably best to add explicit casting here and perhaps an assert to make sure start_offset doesn't have bits past tileOffsets